src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -303,14 +303,19 @@ private Func<object> MakeNewInstance
                     }
                     catch
                     {
-                        _makeNewInstance = () => null;
+                        _makeNewInstance = ReturnNull;
                     }
                 }
 
                 return _makeNewInstance;
             }
         }
 
+        private object ReturnNull()
```

Comment:
Make method static otherwise a closure is created for each instance of ClassDataContract as it needs to capture a reference to `this`.